grafana-stack/deploy.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/sh
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+exec 3>&1
+
+if [[ $# -eq 0 ]]; then
+  modules="grafana influxdb nginx"
+  grafana=1
+  influxdb=1
+  nginx=1
+else
+  grafana=0
+  influxdb=0
+  nginx=0
+  modules="$@"
+
+  for module in "$@"; do
+    case "${module}" in
+      "grafana")
+        grafana=1
+        ;;
+      "influxdb")
+        influxdb=1
+        ;;
+      "nginx")
+        nginx=1
+        ;;
+      *)
+        echo "Unknown module: ${module}"
+        echo "usage: $0 [subdir...]"
+        exit 64
+    esac
+  done
+fi
+
+has() {
+  [[ $1 -ne 0 ]]
+}
+
+print_line() {
+  echo $(date +"%F-%H %T.%N"): "$@" >&3
+
+}
+
+run_command() {
+  what=$1
+  shift
+  print_line ${what}
+  "$@" || {
+    print_line Failed. You can have a look at \"deploy.log\"
+    exit 3
+  }
+}
+
+print_line "Running against modules: ${modules}"
+
+run_command "Reading configuration: deploy.conf" . ./deploy.conf
+print_line "Configuration used:"
+print_line "Kubeconfig:                 ${KUBECONFIG}"
+print_line "Influxdb root password:     ${INFLUXDB_ROOT_PASSWORD}"
+print_line "Grafana admin password:     ${GRAFANA_ADMIN_PASSWORD}"
+print_line "Server host:                ${SERVER_HOST}"
+
+exec 2>deploy.log
+exec 1>&2
+
+set -o xtrace
+
+apply_config() {
+  config=$1
```

Comment:
minor nit: probably safer to declare local variables with `local` so you don't accidentally leak or override